specification/marketplacecatalog/resource-manager/Products/preview/2022-09-01-preview/products.json

```diff
@@ -0,0 +1,1411 @@
+{
+  "x-generator": "NSwag v13.12.1.0 (NJsonSchema v10.4.6.0 (Newtonsoft.Json v13.0.0.0))",
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Marketplace Catalog Browse API",
+    "description": "Microsoft Marketplace Catalog Browse API",
+    "version": "2022-09-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Marketplace/products/{id}": {
+      "get": {
+        "tags": [
+          "SinglePrivateProduct"
+        ],
+        "summary": "Returns a single product directly from BigCat, by subscription as audience",
+        "operationId": "SinglePrivateProduct_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "type": "string",
+            "name": "id",
+            "in": "path",
+            "required": true,
+            "description": "The unique product Id",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "x-ms-partner-tenantid",
+            "x-originalName": "partnerTenantId",
+            "in": "header",
+            "description": "Partner tenant ID",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "x-ms-pricing-audience",
+            "x-originalName": "pricingAudienceHeaderValue",
+            "in": "header",
+            "description": "Pricing audience",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "api-version",
+            "x-originalName": "apiVersion",
+            "in": "query",
+            "required": true,
+            "description": "Version of the REST API used for the request. ",
+            "default": "2022-09-01-preview",
+            "x-nullable": true
+          },
+          {
+            "type": "boolean",
+            "name": "includeStopSoldPlans",
```

Comment:
>includeStopSoldPlans [](http://example.com/codeflow?start=21&length=20)

Instead of boolean parameters, try using enum types. Enums are more future proof and can be extended to larger set of values as needed. e.g. you can make something like: enum named StopSoldPlans with values "Include", "Exclude" for now, and add new values later if needed.

Reply:
Similarly for other boolean parameters in the swagger.

Reply:
Thanks I appreciate your comment and this is indeed a discussion that we had internally but decided to stay with the current implementation of Boolean values